### ISSUE TITLE
CI fails with macos build, even with no code changes

### ISSUE THREAD

**Issue by edwardhartnett** [Unidata/netcdf-c#2974]
This is happening with #2964, which has documentation changes only. So all builds should always work. Yet on MacOS there are failures:

```
73: Test command: /bin/bash "-c" "export srcdir=/Users/runner/work/netcdf-c/netcdf-c/nc_test4;export TOPSRCDIR=/Users/runner/work/netcdf-c/netcdf-c;/Users/runner/work/netcdf-c/netcdf-c/build/nc_test4/test_fillonly.sh "
73: Working Directory: /Users/runner/work/netcdf-c/netcdf-c/build/nc_test4
73: Test timeout computed to be: 1500
36: *** SUCCEED: ref_tst_long_charconst
21: No such file or directory
21: Location: file /Users/runner/work/netcdf-c/netcdf-c/ncdump/nccopy.c; fcn copy line 2010
 62/203 Test  #21: ncdump_tst_nccopy3 ....................***Failed    0.39 sec
```

**Comment by WardF** [comment]
There appears to be a race condition of some sort that was introduced at some point, investigating now.

**Comment by WardF** [comment]
#2964 is merged, looking for the race condition in the failed tests.

**Comment by jrmorgan** [comment]
I'm not sure if you ever tracked down the race condition, but for me, it was occurring because `ncdump/tst_netcdf4.sh` and `ncdump/tst_nccopy4.sh` were both calling `test_ncdump.sh` and hit the nan test at the same time.

**Comment by DennisHeimbigner** [comment]
This line from ncdump/Makefile.am should have fixed this.
````
# Record interscript dependencies so parallel builds work.
tst_nccopy4.log: run_ncgen_tests.log tst_output.log tst_ncgen4.log	\
                 tst_fillbug.log tst_netcdf4_4.log tst_h_scalar.log tst_netcdf4.log
````

**Comment by jrmorgan** [comment]
Yeah I'm not sure, all I know is it printed this:

```
24:
24: *** Testing NaN
24: *** creating NaN test file tst_nans.nc...Sorry! Unexpected result, $SRC_DIR/ncdump/tst_nans.c, line: 63
40:
40: *** Testing NaN
40: *** creating NaN test file tst_nans.nc...ok.
40: *** Tests successful!
```

Therefore I surmised that test 24 was trying to write to `tst_nans.nc` when test 40 hit it.

I'm not an expert at building c packages, and I don't remember exactly what version of the library I'm building - I had put together some custom conda packages last year before 4.9.3 was released to alleviate a memory leak and this built fine on the work machines running linux. I was testing something else on a mac and tried to recreate the packages on that machine. I did check to see if I had that line from the makefile, and I did.
